src/codegen/llvm/llvm_debug_builder.hpp

```diff
@@ -0,0 +1,70 @@
+/*************************************************************************
+ * Copyright (C) 2018-2020 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#pragma once
+
+#include <string>
+
+#include "llvm/IR/DIBuilder.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/Module.h"
+
+namespace nmodl {
+namespace codegen {
+
+/// A struct to store AST location information.
+/// \todo Currently, not all AST nodes have location information. Moreover,
+/// some may not have it as they were artificially introduced (e.g.
+/// CodegenForStatement). This simple wrapper suffices for now, but in future
+/// we may want to handle this properly.
+struct Location {
+    /// Line in the file.
+    int line;
+
+    /// Column in the file.
+    int column;
+};
```

Comment:
to understand - once we add proper `Token` to the CodegenFunction nodes then this won't be needed?

Reply:
I am not sure here. So far I pass proper tokens from `ProcedureBlock` and `FunctionBlock`, so that the lines are known. But if the node is created artificially (e.g. in the kernel), there is no line/column information.

My idea for that class was to have it as a wrapper around location, so that `Token` is used on the lexer/parser level, and for code generation a `Location` wrapper is used (So we separate these two and can add some more functionality to location, such as `loc.getNextLocation()`, etc. Hence, knowing the location of the function start, we can know roughly locations of the inner AST nodes.) But that's not very defined to me yet - so I made this struct for now:)

Reply:
Ok understood. Having bit separation with a wrapper not a bad idea. 👍